### PR TITLE
transform: optimize range over []byte(string)

### PR DESCRIPTION
**Pull request by eliasnaur**

### CODE REVIEW

**Comment by eliasnaur** [comment]
The test failures don't *seem* relevant, but I could easily be mistaken.

**Comment by deadprogram** [comment]
The fix for nix is in... or at least for the CI. Please rebase against latest dev.

**Comment by dgryski** [comment]
LGTM but I really want @aykevl's take on this.

File: transform/util.go [hunk 1]
```diff
@@ -38,15 +38,17 @@ func hasFlag(call, param llvm.Value, kind string) bool {
...
 			if !isReadOnly(use) {
 				return false
 			}
-		} else if !use.IsACallInst().IsNil() {
+		case !use.IsACallInst().IsNil():
 			if !hasFlag(use, value, "readonly") {
 				return false
 			}
-		} else {
+		case !use.IsALoadInst().IsNil():
```
Comment (dgryski) [inline transform/util.go:50]: Maybe add a comment to this block so it doesn't get lost next to the `default` label?

**Review by aykevl — APPROVED** [review]
Yes this looks reasonable (and I agree a comment before `default:` would look slightly better).

File: transform/util.go (hunk 1 above)
Comment (eliasnaur) [reply transform/util.go:50]: Done.

**Review by dgryski — APPROVED** [review]
LGTM

**Pull request merged**